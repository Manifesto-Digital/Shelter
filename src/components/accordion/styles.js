import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../styled/buttons';

export const Wrapper = styled.div`
  position: relative;

  &:not(:first-of-type) {
    &:before {
      position: absolute;
      content: ' ';
      left: 0;
      top: ${({ theme }) => `-${theme.spacing.small}`};
      width: 100%;
      border-top: ${({ theme, active }) =>
        active && `1px solid ${theme.palette.grey45}`};
    }
  }
`;

export const Heading = styled.h3`
  position: relative;
  font-weight: normal;
  color: ${({ theme }) => theme.palette.grey10};
  margin-bottom: ${({ active }) => active && '0'};
`;

export const HeadingButton = styled.button`
  position: relative;
  ${buttonReset};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.xl} ${theme.spacing.standard} ${
      theme.spacing.standard
    }`};
  width: 100%;
  text-align: left;
  font-size: ${({ theme }) => theme.headers.h3};
`;

export const Content = styled.div`
  display: ${({ isOpen }) => (isOpen ? 'block' : 'none')};
  padding: ${({ theme }) =>
    `${theme.spacing.standard} ${theme.spacing.standard} 0 ${
      theme.spacing.standard
    }`};
`;

export const ArrowSVG = styled(SVG)`
  position: absolute;
  transform: ${({ isOpen }) => isOpen && 'rotate(180deg)'} translateY(-50%);
  transform-origin: top;
  top: 50%;
  right: ${({ theme }) => theme.spacing.standard};
  width: 20px;
  height: 25px;

  svg {
    display: block;
    width: 20px;
    height: 25px;
  }
`;
