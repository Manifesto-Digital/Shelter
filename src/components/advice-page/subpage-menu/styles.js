import styled from 'styled-components';

export const MenuNav = styled.nav`
  margin-bottom: ${({ theme }) => theme.spacing.large};

  a {
    text-decoration: none;
  }
`;
