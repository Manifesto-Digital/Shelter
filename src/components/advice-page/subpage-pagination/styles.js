import styled from 'styled-components';
import SVG from 'react-inlinesvg';

export const StyledLink = styled.a`
  text-decoration: none;
  margin-right: ${({ type }) => (type === 'prev' ? 'auto' : null)};
  margin-left: ${({ type }) => (type === 'next' ? 'auto' : null)};

  span {
    text-decoration: underline;
    font-weight: normal;
  }
`;

export const DirectionIcon = styled.div`
  display: flex;
  flex-direction: ${({ type }) => (type === 'prev' ? 'row-reverse' : 'row')};
  justify-content: flex-end;
  font-size: ${({ theme }) => theme.fontsize.larger};
  margin: 0 0 0.4em;
`;

export const Wrapper = styled.div`
  display: flex;
  margin-bottom: ${({ theme }) => theme.spacing.large};
  padding-bottom: ${({ theme }) => theme.spacing.large};
  border-bottom: 2px solid ${({ theme }) => theme.palette.offWhite};
`;

export const StyledSVG = styled(SVG)`
  svg {
    display: block;
    width: 35px;
    height: 40px;
  }
`;
