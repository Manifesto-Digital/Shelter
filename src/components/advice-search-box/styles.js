import styled from 'styled-components';
import Button from '../button';
import { breakpoint } from '../theme/breakpoint';
import TextInput from '../forms/text-input';

export const AdviceSearchForm = styled.form`
  padding: ${props => props.theme.spacing.standard};
  background-color: ${props => props.theme.palette.offWhite};
`;

export const SearchInput = styled(TextInput)`
  margin-bottom: 1em;
  ${breakpoint.tablet`
        display: inline-block;
        width: 75%;
        margin-right: 2em;
    `};
`;

export const SearchButton = styled(Button)`
  background: ${props => props.theme.palette.sanMarinoBlue};
  position: relative;
  width: auto;
  padding: ${props => props.theme.spacing.small};
  color: ${props => props.theme.palette.white};
  display: block;
  ${breakpoint.tablet`
        display: inline-block;
    `};
`;
