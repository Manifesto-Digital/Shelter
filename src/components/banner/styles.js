import styled from 'styled-components'
import LinkHandler from '../link-handler'

export const Wrapper = styled.div`
  padding: ${props => props.theme.spacing.padding} 0;
  background-color: ${props =>
    (props.bg === 'red' && props.theme.palette.primary) ||
    (props.bg === 'blue' && props.theme.palette.sanMarino) ||
    (props.bg === 'black' && props.theme.palette.black) ||
    (props.bg === 'green' && props.theme.palette.donate)};
  margin-bottom: ${props =>
    props.removeMarginBottom ? '0' : props.theme.spacing.largePadding};
`

export const Header = styled.h2`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
export const StyledLinkHandler = styled(LinkHandler)`
  color: ${props =>
    (props.bg === 'red' && props.theme.palette.white) ||
    (props.bg === 'blue' && props.theme.palette.white) ||
    (props.bg === 'black' && props.theme.palette.white) ||
    (props.bg === 'green' && props.theme.palette.white)};
`
