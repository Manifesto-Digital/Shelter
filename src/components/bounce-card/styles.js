import styled from 'styled-components';

export const Wrapper = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.offWhite};
  padding: ${({ theme }) => theme.spacing.standard};
  font-size: ${({ theme }) => theme.fontsize.small};

  img {
    width: 45px;
    height: 76px;
    margin-right: ${({ theme }) => theme.spacing.small};
  }

  div {
    white-space: pre-line;
  }

  p {
    margin-bottom: ${({ theme }) => theme.spacing.small};
  }

  a {
    text-decoration: none;
  }
`;
