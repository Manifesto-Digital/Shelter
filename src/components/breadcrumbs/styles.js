import styled from 'styled-components';
import LinkHandler from '../link-handler';

export const BreadcrumbList = styled.ul`
  margin: 0;
  padding: 0.8em 0;
`;

export const BreadcrumbListItem = styled.li`
  display: inline-block;
  list-style: none;
  margin: 0 0.4em 0 0;
  padding: 0;
  font-size: ${({ theme }) => theme.fontsize.small};

  &:after {
    content: '>';
    margin-left: 0.4em;
  }
  &:last-child {
    &:after {
      content: '';
    }
  }
`;

export const BreadcrumbAnchor = styled(LinkHandler)`
  text-decoration: none;
`;
