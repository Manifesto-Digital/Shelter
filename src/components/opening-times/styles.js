import styled from 'styled-components';
import { breakpoint } from '../theme/breakpoint';

export const TimeList = styled.ul`
  list-style: none;
  padding: 0;
`;

export const Time = styled.li`
  display: flex;
  font-weight: ${({ today }) => today && 'bold'};
  color: ${({ theme, closed }) => closed && theme.palette.grey45};
`;

export const Day = styled.div`
  flex: 1;
`;

export const TimeVal = styled.div`
  ${breakpoint.mobileLand`
    flex: 4;
  `};
`;
