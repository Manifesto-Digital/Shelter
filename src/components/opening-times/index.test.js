import React from 'react';
import { createFactory } from '../../utils/test-factories';
import { snapshotComponent, mountWithTheme } from 'test-helpers';
import OpeningTimes from './index';
import { Time } from './styles';
import theme from '../theme/variables';

jest.mock('../../utils/dates');

export const createOpeningTime = createFactory({
  mondayOpeningTimes: '10am to 6pm',
  tuesdayOpeningTimes: '10am to 6pm',
  wednesdayOpeningTimes: '10am to 6pm',
  thursdayOpeningTimes: '10am to 6pm',
  fridayOpeningTimes: '10am to 6pm',
  saturdayOpeningTimes: '10am to 6pm',
  sundayOpeningTimes: '10am to 6pm',
});

it('renders correctly', () => {
  const mockData = createOpeningTime();

  snapshotComponent(<OpeningTimes data={mockData} />);
});

it('adds a bold style rule to the time on the current day', () => {
  const mockData = createOpeningTime();
  const wrapper = mountWithTheme(<OpeningTimes data={mockData} />);

  // Current day will always be Monday due to the mocked dayOfTheWeek function
  expect(wrapper.find(Time).at(0)).toHaveStyleRule('font-weight', 'bold');
});

it('adds a grey style rule for when a day is listed as closed', () => {
  const mockData = createOpeningTime({
    mondayOpeningTimes: 'Closed',
  });
  const wrapper = mountWithTheme(<OpeningTimes data={mockData} />);

  expect(wrapper.find(Time).at(0)).toHaveStyleRule(
    'color',
    theme.palette.grey45
  );
});
