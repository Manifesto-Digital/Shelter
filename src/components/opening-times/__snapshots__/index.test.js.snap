// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  list-style: none;
  padding: 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: bold;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

@media (min-width:32em) {
  .c3 {
    -webkit-flex: 4;
    -ms-flex: 4;
    flex: 4;
  }
}

<ul
  className="c0"
>
  <li
    className="c1"
  >
    <div
      className="c2"
    >
      Monday:
    </div>
     
    <div
      className="c3"
    >
      10am to 6pm
    </div>
  </li>
  <li
    className="c4"
  >
    <div
      className="c2"
    >
      Tuesday:
    </div>
     
    <div
      className="c3"
    >
      10am to 6pm
    </div>
  </li>
  <li
    className="c4"
  >
    <div
      className="c2"
    >
      Wednesday:
    </div>
     
    <div
      className="c3"
    >
      10am to 6pm
    </div>
  </li>
  <li
    className="c4"
  >
    <div
      className="c2"
    >
      Thursday:
    </div>
     
    <div
      className="c3"
    >
      10am to 6pm
    </div>
  </li>
  <li
    className="c4"
  >
    <div
      className="c2"
    >
      Friday:
    </div>
     
    <div
      className="c3"
    >
      10am to 6pm
    </div>
  </li>
  <li
    className="c4"
  >
    <div
      className="c2"
    >
      Saturday:
    </div>
     
    <div
      className="c3"
    >
      10am to 6pm
    </div>
  </li>
  <li
    className="c4"
  >
    <div
      className="c2"
    >
      Sunday:
    </div>
     
    <div
      className="c3"
    >
      10am to 6pm
    </div>
  </li>
</ul>
`;
